Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -0,0 +1,214 @@
+		raise NotImplementedError()
+
+class _TTGlyphCFF(_TTGlyph):
+	pass
+
+class _TTGlyphGlyf(_TTGlyph):
+
+	def draw(self, pen):
+		"""Draw the glyph onto Pen. See fontTools.pens.basePen for details
+		how that works.
+		"""
+		glyfTable = self._glyphset._glyphs
+		glyph = self._glyph
+		offset = self.lsb - glyph.xMin if hasattr(glyph, "xMin") else 0
+		glyph.draw(pen, glyfTable, offset)
+
+	def drawPoints(self, pen):
+		"""Draw the glyph onto PointPen. See fontTools.pens.pointPen
+		for details how that works.
+		"""
+		glyfTable = self._glyphset._glyphs
+		glyph = self._glyph
+		offset = self.lsb - glyph.xMin if hasattr(glyph, "xMin") else 0
+		glyph.drawPoints(pen, glyfTable, offset)
+
+
+
+class _TTVarGlyphSet(object):
+
+	def __init__(self, font, location, normalized=False):
+		from fontTools.varLib.models import normalizeLocation, piecewiseLinearMap
+		self._ttFont = font
+		if not normalized:
+			axes = {a.axisTag: (a.minValue, a.defaultValue, a.maxValue) for a in font['fvar'].axes}
+			location = normalizeLocation(location, axes)
+			if 'avar' in font:
+				avar = font['avar']
+				avarSegments = avar.segments
+				new_location = {}
+				for axis_tag,value in location.items():
+					avarMapping = avarSegments.get(axis_tag, None)
+					if avarMapping is not None:
+						value = piecewiseLinearMap(value, avarMapping)
+					new_location[axis_tag] = value
+				location = new_location
+				del new_location
+
+		self.location = location
+
+	def keys(self):
+		return list(self._ttFont['glyf'].keys())
+
+	def has_key(self, glyphName):
+		return glyphName in self._ttFont['glyf']
+	__contains__ = has_key
+
+	def __getitem__(self, glyphName):
+		return _TTVarGlyphGlyf(self._ttFont, glyphName, self.location)
+
+	def get(self, glyphName, default=None):
+		try:
+			return self[glyphName]
+		except KeyError:
+			return default
+
+def setCoordinates(glyph, coord, glyfTable):
+	# Handle phantom points for (left, right, top, bottom) positions.
+	assert len(coord) >= 4
+	if not hasattr(glyph, 'xMin'):
+		glyph.recalcBounds(glyfTable)
+	leftSideX = coord[-4][0]
+	rightSideX = coord[-3][0]
+	topSideY = coord[-2][1]
+	bottomSideY = coord[-1][1]
+
+	for _ in range(4):
+		del coord[-1]
+
+	if glyph.isComposite():
+		assert len(coord) == len(glyph.components)
+		for p,comp in zip(coord, glyph.components):
+			if hasattr(comp, 'x'):
+				comp.x,comp.y = p
+	elif glyph.numberOfContours == 0:
+		assert len(coord) == 0
+	else:
+		assert len(coord) == len(glyph.coordinates)
+		glyph.coordinates = coord
+
+	glyph.recalcBounds(glyfTable)
+
+	horizontalAdvanceWidth = rightSideX - leftSideX
+	verticalAdvanceWidth = topSideY - bottomSideY
+	leftSideBearing = glyph.xMin - leftSideX
+	return horizontalAdvanceWidth, leftSideBearing, verticalAdvanceWidth
+
+
+class _TTVarGlyphGlyf(object):
+
+	def __init__(self, ttFont, glyphName, location):
+		self._ttFont = ttFont
+		self._glyphName = glyphName
+		self._location = location
+		#self.width, self.lsb = ttFont['hmtx'][glyphName]
```

Comment:
maybe initialize these attributes to None or 0, and comment that the get updated once you draw the glyph.